macOSUpgrade.sh

```diff
@@ -199,13 +200,18 @@ downloadInstaller() {
 
 get_power_status() {
     ##Check if device is on battery or ac power
-    ##If not allow user to connect to power for specified time period
+    ##If not, and our acPowerWaitTimer is above 1, allow user to connect to power for specified time period
     if /usr/bin/pmset -g ps | grep "AC Power" > /dev/null ; then
         /bin/echo "Power Check: OK - AC Power Detected"
     else
-        /Library/Application\ Support/JAMF/bin/jamfHelper.app/Contents/MacOS/jamfHelper -windowType utility -title "Waiting for AC Power Connection" -icon "$warnIcon" -description "Please connect your computer to power using an AC power adapter. This process will continue once AC power is detected." &
-        jamfHelperPowerPID=$(/bin/echo $!)
-        wait_for_ac_power
+        if [[ -n "$acPowerWaitTimer" ]] && [[ "$acPowerWaitTimer" -gt 0 ]]; then
```

Comment:
It would be make simple like that,
`if [[ "${$acPowerWaitTimer:-0}" -gt 0 ]]; then`

Reply:
`if [[ "${$acPowerWaitTimer:-0}" -gt 0 ]]; then` gives me an error. I've never seen a comparison test like that. However, I replaced line with `f [[ "$acPowerWaitTimer" -gt 0 ]]; then` as that should ensure the variable is not null and an integer greater than 0.